daemon/src/rollover/protocol.rs

```diff
@@ -141,3 +190,402 @@ pub(crate) async fn emit_failed(order_id: OrderId, e: anyhow::Error, executor: &
+
+    let lock_tx = PartiallySignedTransaction::from_unsigned_tx(unsigned_lock_tx)?;
+    let own_cfd_txs = tokio::task::spawn_blocking({
+        let maker_address = dlc.maker_address.clone();
+        let taker_address = dlc.taker_address.clone();
+        let lock_tx = lock_tx.clone();
+
+        move || {
+            renew_cfd_transactions(
+                lock_tx,
+                (
+                    maker_identity.into(),
+                    maker_lock_amount,
+                    maker_address,
+                    maker_punish_params,
+                ),
+                (
+                    taker_identity.into(),
+                    taker_lock_amount,
+                    taker_address,
+                    taker_punish_params,
+                ),
+                oracle_pk,
+                (CET_TIMELOCK, rollover_params.refund_timelock),
+                payouts,
+                sk.into(),
+                rollover_params.fee_rate.to_u32(),
+            )
+        }
+    })
+    .await?
+    .context("Failed to create new CFD transactions")?;
+
+    Ok(own_cfd_txs)
+}
+
+pub(crate) fn build_commit_descriptor(
+    ((maker_identity, maker_punish_params), (taker_identity, taker_punish_params)): (
+        (PublicKey, PunishParams),
+        (PublicKey, PunishParams),
+    ),
+) -> Descriptor<bdk::bitcoin::PublicKey> {
+    commit_descriptor(
+        (
+            maker_identity.into(),
+            maker_punish_params.revocation_pk,
+            maker_punish_params.publish_pk,
+        ),
+        (
+            taker_identity.into(),
+            taker_punish_params.revocation_pk,
+            taker_punish_params.publish_pk,
+        ),
+    )
+}
+
+#[allow(clippy::too_many_arguments)]
+pub(crate) async fn build_and_verify_cets_and_refund(
+    dlc: &Dlc,
+    announcement: &olivia::Announcement,
+    oracle_pk: schnorrsig::PublicKey,
+    publish_pk: bdk::bitcoin::PublicKey,
+    our_role: Role,
+    own_cfd_txs: &CfdTransactions,
+    commit_desc: &Descriptor<bdk::bitcoin::PublicKey>,
+    msg1: &RolloverMsg1,
+) -> Result<(HashMap<BitMexPriceEventId, Vec<Cet>>, Transaction)> {
+    let lock_amount = dlc.taker_lock_amount + dlc.maker_lock_amount;
+
+    // TODO: Potentially optimize clone by using references and dereferencing smarter when creating
+    // CETs
+    let own_cets = own_cfd_txs.cets.clone();
+    let commit_tx = own_cfd_txs.commit.0.clone();
+
+    let commit_amount = Amount::from_sat(commit_tx.output[0].value);
+
+    verify_adaptor_signature(
+        &commit_tx,
+        &dlc.lock.1,
+        lock_amount,
+        &msg1.commit,
+        &publish_pk,
+        &dlc.identity_counterparty.into(),
+    )
+    .context("Commit adaptor signature does not verify")?;
+
+    let other_address = match our_role {
+        Role::Maker => dlc.taker_address.clone(),
+        Role::Taker => dlc.maker_address.clone(),
+    };
+
+    for own_grouped_cets in own_cets.clone() {
+        let other_cets = msg1
```

Comment:
Is this cone needed?

Reply:
I tried to make this more explicit, see https://github.com/itchysats/itchysats/pull/2248/commits/690e63b02f4b5b3f3cd2a3088f196af6391f8a2f